packages/embark-mocha-tests/src/lib/index.js

```diff
@@ -103,7 +104,13 @@ class MochaTestRunner {
         events.request("contracts:reset", next);
       },
       (next) => { // get contract files
-        events.request("config:contractsFiles", next);
+        if (!options.coverage) {
+          return events.request("config:contractsFiles", next);
+        }
+
+        events.request("coverage:prepareContracts", () => {
```

Comment:
as discussed, let's remove it from here, and add a `before` and `after` actions to the test runner which the coverage plugin can use